openedx/core/djangoapps/user_api/api.py

```diff
@@ -419,22 +417,11 @@ def _add_password_field(self, form_desc, required=True):
         # meant to hold the user's password.
         password_label = _(u"Password")
 
-        restrictions = {
-            "min_length": password_min_length(),
-            "max_length": password_max_length(),
-        }
-
-        complexities = password_complexity()
-        for key, value in complexities.iteritems():
-            api_key = key.lower().replace(' ', '_')
-            restrictions[api_key] = value
-
         form_desc.add_field(
             "password",
             label=password_label,
             field_type="password",
-            instructions=password_instructions(),
-            restrictions=restrictions,
+            instructions=password_validators_instruction_texts(),
```

Comment:
These encoded restrictions are used by the mobile apps to do app-side validation. We would need a solution for them here, rather than just dropping this info. (Similar problem to the old login registration page issue.)

Reply:
Write a function similar to get_instruction_texts() to construct this dictionary of complexity restrictions